R/FilterStateDate.R

```diff
@@ -343,12 +343,33 @@ DateFilterState <- R6::R6Class( # nolint
               logger::log_trace("DateFilterState$server@2 selection changed, id: { private$get_id() }")
               start_date <- input$selection[1]
               end_date <- input$selection[2]
+
+              if (is.na(start_date)) {
+                start_date <- private$get_selected()[1]
+                updateDateRangeInput(
+                  session = session,
+                  inputId = "selection",
+                  start = start_date
+                )
+                showNotification("Start date must be selected. Setting back to previous value.", type = "warning")
+              }
+
+              if (is.na(end_date)) {
+                end_date <- private$get_selected()[2]
+                updateDateRangeInput(
+                  session = session,
+                  inputId = "selection",
+                  end = end_date
+                )
+                showNotification("End date must be selected. Setting back to previous value.", type = "warning")
+              }
+
               if (start_date > end_date) {
-                showNotification(
-                  "Start date must not be greater than the end date. Setting back to default values.",
+                showNotification("Start date must not be greater than the end date. Setting back to default values.",
                   type = "warning"
                 )
               }
+
```

Comment:
previous solution `start_date > end_date` was wrong as it just showed notification and sent invalid `start_date` and `end_date` to set selected (without escaping operation). So I suggest to combine it all in one condition to avoid writing updateRangeInput multiple times.

```suggestion
              if (is.na(start_date) || is.na(end_date) || start_date > end_date) {
                updateDateRangeInput(
                  session = session,
                  inputId = "selection",
                  start = private$get_selected()[1],
                  end = private$get_selected()[2]
                )
                showNotification(
                  "Start date must not be greater than the end date. Setting back to previous value.",
                  type = "warning"
                )
                return(NULL)
              }
              
```

Reply:
@gogonzo! Are the same notifications applicable to all three conditions? I believe they can be comprehended for each of the three scenarios.